pha-admin/src/main/java/com/store/pharmacy/Company/serviceImpl/CompanyServiceImpl.java

```diff
@@ -0,0 +1,91 @@
+package com.store.pharmacy.Company.serviceImpl;
+
+import com.store.pharmacy.Company.model.Company;
+import com.store.pharmacy.Company.model.CompanyInput;
+import com.store.pharmacy.Company.model.CompanyOutput;
+import com.store.pharmacy.Company.repository.CompanyRepository;
+import com.store.pharmacy.Company.service.CompanyService;
+import com.store.pharmacy.exception.DataNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+
+@Service
+@Transactional
+public class CompanyServiceImpl implements CompanyService {
+    @Autowired
+    CompanyRepository companyRepository;
+
+    private final String INSERT = "INSERT";
+    private final String UPDATE = "UPDATE";
+
+    @Override
+    public CompanyOutput save(CompanyInput companyInput) {
+        Company inParam = convertCompanyInputToCompany(null, companyInput, INSERT);
+        Company savedCompany = companyRepository.save(inParam);
+        CompanyOutput outParam = convertCompanyToCompanyOutput(savedCompany, INSERT);
+        return outParam;
+    }
+
+    @Override
+    public CompanyOutput update(String idCompany, CompanyInput companyInput) {
+        long idInput;
+        try {
+            idInput = Long.parseLong(idCompany);
+        } catch (NumberFormatException ex) {
+            throw new DataNotFoundException();
+        }
+        Company company = companyRepository.findById(idInput).get();
+        if (company == null) {
+            throw new DataNotFoundException();
+        }
+        Company inParam = convertCompanyInputToCompany(company, companyInput, UPDATE);
+        Company savedCompany = companyRepository.save(inParam);
+        CompanyOutput outParam = convertCompanyToCompanyOutput(savedCompany, UPDATE);
```

Comment:
Immediately return this expression instead of assigning it to the temporary variable "outParam".